js/libs/builder.js

```diff
@@ -299,29 +299,37 @@ var app = new Vue({
           };
 
           Fliplet.DataSources.getById($vm.settings.dataSourceId).then(function(dataSource) {
-            if (dataSource.hooks.length) {
-              // Update existing hook
-              var currentHook = _.find(dataSource.hooks, function(o) {
-                return o.widgetInstanceId == $vm.settings.id;
-              });
-
-              currentHook.payload = $vm.settings.emailTemplateAdd;
-
-              var index = _.findIndex(dataSource.hooks, function(o) {
-                return o.widgetInstanceId == $vm.settings.id;
-              });
-              dataSource.hooks.splice(index, 1, currentHook);
-
-              Fliplet.DataSources.update($vm.settings.dataSourceId, {
+            if (!dataSource.hooks.length) {
+              // No hooks found at all. Add new hook.
+              dataSource.hooks.push(newHook);
+              return Fliplet.DataSources.update($vm.settings.dataSourceId, {
                 hooks: dataSource.hooks
               });
-            } else {
-              // Add new hook
+            }
+
+            // Update existing hook
+            var currentHook = _.find(dataSource.hooks, function(o) {
+              return o.widgetInstanceId == $vm.settings.id;
+            });
+            
+            if (!currentHook) {
```

Comment:
I've assessed the issue as much as I can and it appears that `$vm.settings.id` is `undefined` hence `currentHook` is not found.

This is a problem because even if new hooks are added they wouldn't have a `widgetInstanceId` reference.

Question is: why would `$vm.settings.id` be `undefined`?

Reply:
@squallstar Any idea?

Reply:
uhm, I think @hcarneiro changed this code since I first implemented it. I can't remember of `$vm.settings.id`. @hcarneiro does it ring any bell?

@tonytlwu I'm happy to take a look this but I don't think I will be able to before Monday.